### PR TITLE
Story 1 into main

### PR DESCRIPTION
**Pull request by chloe-smart**

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+<html>
+<head>
+    <title>Chloe's Collector App</title>
+    <link rel="stylesheet" href="https://use.typekit.net/rkk2hbc.css">
+    <link rel="stylesheet" type="text/css" href="styles/style.css">
+    <html lang="en"></html>
+</head>
+    <header>
+        <h1 tabindex=0 >ARTISTS</h1>
+    </header>
```
Comment (ashleycoles) [inline index.php:22]: This should be within the body tag - all content visible on the page should live within the body tag.

File: index.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+<body>
+    <?php
+    foreach($artists as $artist) {
+        try {
+            $artistsOutput = generateArtistHtml($artist);
+            echo $artistsOutput;
```
Comment (ashleycoles) [inline index.php:28]: This isn't a change request, this bit of code is totally fine as it is.

Some would say that creating the variable here isn't needed as you're only using the results of `generateArtistHtml()` a single time - you could just do `echo generateArtistHtml($artist);`

Like I said, not something you really need to change, I'd accept it as is, just one to note!

**Review by ashleycoles — CHANGES_REQUESTED** [review]
A few changes for you, all very minor things! Really nice work :)

File: index.php (hunk 1 above)
Comment (chloe-smart) [reply index.php:22]: done🙂

File: index.php (hunk 2 above)
Comment (chloe-smart) [reply index.php:28]: Changed to one line!

File: src/functions.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+
+function generateArtistHtml(array $artists): string {
+    if (array_key_exists('name', $artists) 
+    && array_key_exists('yearsLived', $artists) 
+    && array_key_exists('image', $artists) 
+    && array_key_exists('favouriteMedium', $artists)
+    && array_key_exists('knownFor', $artists)
+    && array_key_exists('placeOfBirth', $artists)) {
```
[the remark this answers is not in the document]
Comment (chloe-smart) [reply src/functions.php:10]: done🙂

File: src/functions.php [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        return '<h2 tabindex=0 >' . $artists['name'] . '</h2>' . 
+        '<p tabindex=0 class="years">' . $artists['yearsLived'] . '</p>' .
+        '<img tabindex=0 src="images/' . $artists['image'] . ' "alt="an image of artwork by ' . $artists['name'] . '">' .
+        '<h3 tabindex=0 >Favourite Medium: ' . $artists['favouriteMedium'] . '</h3>' . 
+        '<h3 tabindex=0 >Known For: ' . $artists['knownFor'] . '</h3>' . 
+        '<h3 tabindex=0 >Place Of Birth: ' . $artists['placeOfBirth'] . '</h3>';
+    } else {
+        throw new Exception('Missing or incorrect array keys entered'); 
+    }
```
[the remark this answers is not in the document]
Comment (chloe-smart) [reply src/functions.php:19]: Done, I think!

File: src/test/functions.php [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+require '../functions.php';
+
+use PHPUnit\Framework\TestCase;
+
+class Functions extends TestCase
+{
+    public function testGenerateArtistHtmlSuccess()
+    {
+        $artistPlaceholder = ['name' => 'Salvador Dali',
```
[the remark this answers is not in the document]
Comment (chloe-smart) [reply src/test/functions.php:11]: done

File: styles/style.css [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+    img {
+        width: 300px;
+        height: 300px;
+    }
+}
+
+@media screen and (max-width: 450px) {
+    h1 {
+        letter-spacing: 0px;
```
[the remark this answers is not in the document]
Comment (chloe-smart) [reply styles/style.css:74]: done🙂

File: src/db.php [hunk 7]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+function connectToDb(string $dbName): PDO {
+	$db = new PDO("mysql:host=db; dbname=$dbName", 'root', 'password');
+	$db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+	return $db;
+}
+
+function getAllArtists(PDO $db): array {
+	$query = $db->prepare('SELECT * FROM artists;');
```
[the remark this answers is not in the document]
Comment (chloe-smart) [reply src/db.php:10]: sorted🙃

**Review by ashleycoles — APPROVED** [review]
Spot on! Nicely done :)

**Pull request merged**